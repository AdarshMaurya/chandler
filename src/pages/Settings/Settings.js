import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {Text, Switch, View, Platform} from 'react-native';
import {Button, Navbar} from 'components';

import {commonStyles, primaryColor} from 'theme';
import {styles} from './Settings.styles';

export class Settings extends PureComponent {
  static propTypes = {
    logout: PropTypes.func.isRequired,
    goToLaunchScreen: PropTypes.func.isRequired,
    beta: PropTypes.bool.isRequired,
    subscribeBeta: PropTypes.func.isRequired,
  };

  logout = () => {
    this.props.logout();
    this.props.goToLaunchScreen();
  };

  render() {
    const {beta, subscribeBeta} = this.props;
    return (
      <View style={commonStyles.flex}>
        <Navbar title="Settings" />

        <View style={[commonStyles.flex, styles.container]}>
          <View style={styles.betaContainer}>
            <View style={styles.rowCentered}>
              <Switch
                value={beta}
                onValueChange={subscribeBeta}
                thumbTintColor={
                  (Platform.OS === 'android' && primaryColor) || null
                }
              />
              <Text style={styles.betaTitle}>Early Beta access.</Text>
            </View>

            <Text style={styles.betaSubtitle}>
              Please be careful ! You phone might automatically install a buggy
              version of Monica on your phone.
            </Text>
          </View>

          <Button onPress={this.logout}>Logout</Button>
        </View>
      </View>
    );
  }
}
