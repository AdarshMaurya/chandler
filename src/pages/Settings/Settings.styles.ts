import {StyleSheet} from 'react-native';

import {mutedTextColor} from '@theme';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#E2E4E5',
  },
  betaToggleContainer: {
    backgroundColor: 'white',
    paddingVertical: 15,
    paddingHorizontal: 20,
    flexDirection: 'row',
    alignItems: 'center',
  },
  betaTitle: {
    fontSize: 14,
    flex: 1,
  },
  betaInfo: {
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  betaSubtitle: {
    fontSize: 12,
  },
  logoutContainer: {
    backgroundColor: 'white',
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  logoutText: {
    color: '#1361cb',
  },
  appVersion: {
    color: mutedTextColor,
    fontSize: 12,
    marginTop: 10,
  },
  freepik1: {
    color: mutedTextColor,
    textAlign: 'center',
    fontSize: 10,
    marginTop: 10,
  },
  freepik2: {
    color: mutedTextColor,
    textAlign: 'center',
    fontSize: 11,
    marginBottom: 10,
    fontWeight: '600',
  },
});
