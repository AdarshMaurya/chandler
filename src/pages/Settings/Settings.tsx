// tslint:disable-next-line:no-var-requires
const {version} = require('../../../package.json');
import React, {PureComponent} from 'react';
import {Platform, Switch, Text, TouchableOpacity, View} from 'react-native';

import {Navbar} from '@components';
import {I18n} from '@i18n';
import {
  IBetaSubscribeOperation,
  IRouterGoToLaunchScreenOperation,
  IUserLogoutOperation,
} from '@models/operations';
import {commonStyles, primaryColor} from '@theme';
import {styles} from './Settings.styles';

interface ISettingsProps {
  beta: boolean;
  subscribeBeta: IBetaSubscribeOperation;
  logout: IUserLogoutOperation;
  goToLaunchScreen: IRouterGoToLaunchScreenOperation;
}

export class Settings extends PureComponent<ISettingsProps, {}> {
  public logout = () => {
    this.props.logout();
    this.props.goToLaunchScreen();
  };

  public render() {
    const {beta, subscribeBeta} = this.props;
    return (
      <View style={commonStyles.flex}>
        <Navbar title={I18n.t('settings:settings')} />

        <View style={[commonStyles.flex, styles.container]}>
          <View style={styles.betaToggleContainer}>
            <Text style={styles.betaTitle}>
              {I18n.t('settings:betaInstall')}
            </Text>
            <Switch
              value={beta}
              onValueChange={subscribeBeta}
              thumbTintColor={
                (Platform.OS === 'android' && primaryColor) || null
              }
            />
          </View>

          <View style={styles.betaInfo}>
            <Text style={styles.betaSubtitle}>
              {I18n.t('settings:betaInstallExplanation')}
            </Text>
            <Text style={styles.appVersion}>
              {I18n.t('settings:version', {version})}
            </Text>
          </View>

          <TouchableOpacity onPress={this.logout}>
            <View style={styles.logoutContainer}>
              <Text style={styles.logoutText}>{I18n.t('settings:logout')}</Text>
            </View>
          </TouchableOpacity>

          <View style={commonStyles.flex} />
          <Text style={styles.freepik1}>{I18n.t('settings:freepik1')}</Text>
          <Text style={styles.freepik2}>{I18n.t('settings:freepik2')}</Text>
        </View>
      </View>
    );
  }
}
