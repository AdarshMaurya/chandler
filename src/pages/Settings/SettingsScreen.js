import {connect} from 'react-redux';
import {logout} from 'redux/user';
import {goToLaunchScreen} from 'redux/router';
import {subscribeBeta} from 'redux/beta';

import {Settings} from './Settings';

export const mapStateToProps = state => ({
  beta: state.beta.isSubscribed,
});

export const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(logout()),
  goToLaunchScreen: () => dispatch(goToLaunchScreen()),
  subscribeBeta: isSubscribed => dispatch(subscribeBeta(isSubscribed)),
});

export const SettingsScreen = connect(mapStateToProps, mapDispatchToProps)(
  Settings,
);
