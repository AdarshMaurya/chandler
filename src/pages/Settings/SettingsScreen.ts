import {subscribeBeta} from '@redux/beta';
import {goToLaunchScreen} from '@redux/router';
import {logout} from '@redux/user';
import {connect} from 'react-redux';

import {Settings} from './Settings';

export const mapStateToProps = state => ({
  beta: state.beta.isSubscribed,
});

export const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(logout()),
  goToLaunchScreen: () => dispatch(goToLaunchScreen()),
  subscribeBeta: isSubscribed => dispatch(subscribeBeta(isSubscribed)),
});

export const SettingsScreen = connect(mapStateToProps, mapDispatchToProps)(
  Settings
);
