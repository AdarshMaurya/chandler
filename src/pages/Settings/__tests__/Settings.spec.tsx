import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Settings} from '../Settings';

describe('Pages', () => {
  describe('Settings', () => {
    it('ios should renders correctly', () => {
      const logout = jest.fn();
      const goToLaunchScreen = jest.fn();
      const subscribeBeta = jest.fn();

      expect(
        toJson(
          shallow(
            <Settings
              logout={logout}
              goToLaunchScreen={goToLaunchScreen}
              subscribeBeta={subscribeBeta}
              beta={true}
            />
          )
        )
      ).toMatchSnapshot();

      expect(
        toJson(
          shallow(
            <Settings
              logout={logout}
              goToLaunchScreen={goToLaunchScreen}
              subscribeBeta={subscribeBeta}
              beta={false}
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('android should renders correctly', () => {
      const logout = jest.fn();
      const goToLaunchScreen = jest.fn();
      const subscribeBeta = jest.fn();

      jest.mock('Platform', () => ({
        OS: 'android',
      }));

      expect(
        toJson(
          shallow(
            <Settings
              logout={logout}
              goToLaunchScreen={goToLaunchScreen}
              subscribeBeta={subscribeBeta}
              beta={true}
            />
          )
        )
      ).toMatchSnapshot();

      expect(
        toJson(
          shallow(
            <Settings
              logout={logout}
              goToLaunchScreen={goToLaunchScreen}
              subscribeBeta={subscribeBeta}
              beta={false}
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('logout should call logout and go to launch screen', () => {
      const logout = jest.fn();
      const goToLaunchScreen = jest.fn();
      const subscribeBeta = jest.fn();

      const tree = shallow(
        <Settings
          logout={logout}
          goToLaunchScreen={goToLaunchScreen}
          subscribeBeta={subscribeBeta}
          beta={true}
        />
      );

      tree.instance().logout();
      expect(logout.mock.calls.length).toBe(1);
      expect(logout.mock.calls[0]).toEqual([]);
      expect(goToLaunchScreen.mock.calls.length).toBe(1);
      expect(goToLaunchScreen.mock.calls[0]).toEqual([]);
    });
  });
});
