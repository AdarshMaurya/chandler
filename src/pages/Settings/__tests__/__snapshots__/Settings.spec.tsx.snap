// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Settings android should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Navbar
    title="Settings"
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#E2E4E5",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 14,
          }
        }
      >
        Enable automatic beta installs.
      </Text>
      <Switch
        disabled={false}
        onValueChange={[MockFunction]}
        thumbTintColor="#355974"
        value={true}
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        If enabled, your application will automatically install beta versions. That will let you test cutting edge features, but beware though: you might install versions that are not stable yet.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#676767",
            "fontSize": 12,
            "marginTop": 10,
          }
        }
      >
        Version: 1.0.0
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1361cb",
            }
          }
        >
          Logout
        </Text>
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Most illustrations used in this app come from Freepik
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 11,
          "fontWeight": "600",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Designed by freepik.com
    </Text>
  </View>
</View>
`;

exports[`Pages Settings android should renders correctly 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Navbar
    title="Settings"
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#E2E4E5",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 14,
          }
        }
      >
        Enable automatic beta installs.
      </Text>
      <Switch
        disabled={false}
        onValueChange={[MockFunction]}
        thumbTintColor="#355974"
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        If enabled, your application will automatically install beta versions. That will let you test cutting edge features, but beware though: you might install versions that are not stable yet.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#676767",
            "fontSize": 12,
            "marginTop": 10,
          }
        }
      >
        Version: 1.0.0
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1361cb",
            }
          }
        >
          Logout
        </Text>
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Most illustrations used in this app come from Freepik
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 11,
          "fontWeight": "600",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Designed by freepik.com
    </Text>
  </View>
</View>
`;

exports[`Pages Settings ios should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Navbar
    title="Settings"
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#E2E4E5",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 14,
          }
        }
      >
        Enable automatic beta installs.
      </Text>
      <Switch
        disabled={false}
        onValueChange={[MockFunction]}
        thumbTintColor={null}
        value={true}
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        If enabled, your application will automatically install beta versions. That will let you test cutting edge features, but beware though: you might install versions that are not stable yet.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#676767",
            "fontSize": 12,
            "marginTop": 10,
          }
        }
      >
        Version: 1.0.0
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1361cb",
            }
          }
        >
          Logout
        </Text>
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Most illustrations used in this app come from Freepik
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 11,
          "fontWeight": "600",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Designed by freepik.com
    </Text>
  </View>
</View>
`;

exports[`Pages Settings ios should renders correctly 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Navbar
    title="Settings"
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#E2E4E5",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 14,
          }
        }
      >
        Enable automatic beta installs.
      </Text>
      <Switch
        disabled={false}
        onValueChange={[MockFunction]}
        thumbTintColor={null}
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        If enabled, your application will automatically install beta versions. That will let you test cutting edge features, but beware though: you might install versions that are not stable yet.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#676767",
            "fontSize": 12,
            "marginTop": 10,
          }
        }
      >
        Version: 1.0.0
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "paddingHorizontal": 20,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1361cb",
            }
          }
        >
          Logout
        </Text>
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Most illustrations used in this app come from Freepik
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 11,
          "fontWeight": "600",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Designed by freepik.com
    </Text>
  </View>
</View>
`;
